# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1295214379 -3600
#      Sun Jan 16 22:46:19 2011 +0100
# Node ID beb4f0f27a3299e61254e5214f7c41e360a80f4f
# Parent  50a7935f25122c94ecde47007013ca64d4d2f15f
Use {} as the default font for the gnuplot backend

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,11 +1,15 @@
+2011-01-16  David Bateman  <dbateman@free.fr>
+
+	* plot/__go_draw_axes__.m: Use "{}" as the default font.
+
 2011-01-15  Rik  <octave@nomad.inbox5.com>
 
-	* scripts/help/doc.m, scripts/help/which.m, 
+	* scripts/help/doc.m, scripts/help/which.m,
 	scripts/miscellaneous/comma.m, scripts/miscellaneous/ls.m,
 	scripts/miscellaneous/paren.m, scripts/miscellaneous/semicolon.m,
 	scripts/pkg/pkg.m: Eliminate @deffn macros.
 
 2011-01-15  John W. Eaton  <jwe@octave.org>
 
 	* deprecated/krylovb: Move here from linear-algebra.
 	* deprecated/module.mk (deprecated_FCN_FILES):
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -1467,17 +1467,18 @@ function __go_draw_axes__ (h, plot_strea
           pos = "";
           warning ("legend: 'Best' not yet implemented for location specifier.\n");
           ## Least conflict with data in plot.
           ## Least unused space outside plot.
         otherwise
           pos = "";
       endswitch
       if (__gnuplot_has_feature__ ("key_has_font_properties"))
-        fontspec = create_fontspec (hlgnd.fontname, hlgnd.fontsize, gnuplot_term);
+        [fontname, fontsize] = get_fontname_and_size (obj);
+        fontspec = create_fontspec (fontname, fontsize, gnuplot_term);
       else
         fontspec = "";
       endif
       fprintf (plot_stream, "set key %s %s;\nset key %s %s %s %s;\n", 
                inout, pos, box, reverse, horzvert, fontspec);
     else
       fputs (plot_stream, "unset key;\n");
     endif
@@ -2077,18 +2078,18 @@ function colorspec = get_text_colorspec 
     colorspec = "";
   else
     colorspec = sprintf ("textcolor rgb \"#%02x%02x%02x\"",
                          round (255*color));
   endif
 endfunction
 
 function [f, s, fnt, it, bld] = get_fontname_and_size (t)
-  if (isempty (t.fontname))
-    fnt = "Helvetica";
+  if (isempty (t.fontname) || strcmp (t.fontname, "*"))
+    fnt = "{}";
   else
     fnt = t.fontname;
   endif
   f = fnt;
   it = false;
   bld = false;
   if (! isempty (t.fontweight) && strcmpi (t.fontweight, "bold"))
     if (! isempty(t.fontangle)
